container_linux.go

```diff
@@ -249,6 +257,113 @@ func (c *linuxContainer) NotifyOOM() (<-chan struct{}, error) {
 	return notifyOnOOM(c.cgroupManager.GetPaths())
 }
 
+// XXX debug support, remove when debugging done.
+func addArgsFromEnv(evar string, args *[]string) {
+	if e := os.Getenv(evar); e != "" {
+		for _, f := range strings.Fields(e) {
+			*args = append(*args, f)
+		}
+	}
+	fmt.Printf(">>> criu %v\n", *args)
+}
+
+func (c *linuxContainer) Checkpoint() error {
+	c.m.Lock()
+	defer c.m.Unlock()
+	dir := filepath.Join(c.root, "checkpoint")
+	// Since a container can be C/R'ed multiple times,
+	// the checkpoint directory may already exist.
+	if err := os.Mkdir(dir, 0655); err != nil && !os.IsExist(err) {
+		return err
+	}
+	args := []string{
```

Comment:
Most of the flags have been there for many years and we haven't introduced new flags in a long time.  But if someone is running an old version of CRIU that doesn't support a specific flag, CRIU will print an error message and exit -- no damage done.  That said, we can certainly add code to do version check every time but I really think it'd be an overkill :)